functional/FunctionalTest.java

```diff
@@ -3823,12 +3836,112 @@ public static Process runMinio() throws Exception {
     pb.redirectErrorStream(true);
     pb.redirectOutput(ProcessBuilder.Redirect.to(new File(MINIO_BINARY + ".log")));
 
-    System.out.println("starting minio server");
+    if (tls) {
+      System.out.println("starting minio server in TLS");
+    } else {
+      System.out.println("starting minio server");
+    }
     Process p = pb.start();
     Thread.sleep(10 * 1000); // wait for 10 seconds to do real start.
     return p;
   }
 
+  public static void runAutomatedEndpointTests() throws Exception {
+    // Run non-TLS tests on plain HTTP endpoint
+    client = MinioClient.builder().endpoint(endpoint).credentials(accessKey, secretKey).build();
+    MinioAdminClient adminClient =
+        MinioAdminClient.builder().endpoint(endpoint).credentials(accessKey, secretKey).build();
+    adminClientTests = new TestMinioAdminClient(adminClient, mintEnv);
+    // Enable trace for debugging.
+    // client.traceOn(System.out);
+    if (!mintEnv) System.out.println(">>> Running tests:");
+    FunctionalTest.runTests();
+
+    isSecureEndpoint = true;
+
+    // Run TLS tests on TLS endpoint
+    client = MinioClient.builder().endpoint(endpointTLS).credentials(accessKey, secretKey).build();
+    client.ignoreCertCheck();
+    adminClient =
+        MinioAdminClient.builder()
+            .endpoint(endpointTLS)
+            .credentials(accessKey, secretKey)
+            .httpClient(getUnsafeOkHttpClient())
+            .build();
+    adminClientTests = new TestMinioAdminClient(adminClient, mintEnv);
+    // Enable trace for debugging.
+    // client.traceOn(System.out);
+    if (!mintEnv) System.out.println(">>> Running TLS tests:");
+    FunctionalTest.runTests();
+
+    isSecureEndpoint = false;
+
+    if (!mintEnv) {
+      System.out.println();
+      System.out.println(">>> Running tests for region:");
+      isQuickTest = true;
+      // Get new bucket name to avoid minio azure gateway failure.
+      bucketName = getRandomName();
+      bucketNameWithLock = getRandomName();
+      client =
+          MinioClient.builder()
+              .endpoint(endpoint)
+              .credentials(accessKey, secretKey)
+              .region(region)
+              .build();
+      adminClient =
+          MinioAdminClient.builder()
+              .endpoint(endpoint)
+              .credentials(accessKey, secretKey)
+              .region(region)
+              .build();
+      adminClientTests = new TestMinioAdminClient(adminClient, mintEnv);
+      FunctionalTest.runTests();
+    }
+  }
+
+  public static void runCustomEndpointTests() throws Exception {
+    isSecureEndpoint = endpoint.toLowerCase(Locale.US).contains("https://");
+    client = MinioClient.builder().endpoint(endpoint).credentials(accessKey, secretKey).build();
+    client.ignoreCertCheck();
```

Comment:
No need to do this for endpoint argument passed. User knows how to use functional test in this case.